### PR TITLE
Include search term in title for improved accessibility

### PR DESCRIPTION
**Pull request by metamoni**
## What type of PR is this? (check all applicable)

Not sure. Feature? 🤷🏻‍♀️
Maybe an "accessibility" category would work here

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR aims to add the search term to the page title (`h1`, not meta title), as per recommendations from Deque and accessibility best practices.

## Related Tickets & Documents
#14735 

## QA Instructions, Screenshots, Recordings
- go to your dashboard
- type something into the search input and hit Enter
- verify that the page title reads "Search results for [your search term]"
- delete your search term and hit Enter again
- verify that the page title reads "Search results"

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

**Comment by metamoni** [comment]
Not sure why this appears as a merge conflict, rather than just a file change 🤔

**Comment by djuber** [comment]
> Not sure why this appears as a merge conflict, rather than just a file change thinking


```html
  <h1 class="crayons-title hidden s:block pl-2">Search results</h1>
```

It looks like the merge conflict is tied to the main branch having additional classes on the h1 element `(hidden s:block pl-2`) but this branch not including those.

**Review by nickytonline — APPROVED** [review]
Thanks for making search more accessible @metamoni! 🚀

**Comment by citizen428** [comment]
I resolved the merge conflict for you @metamoni, thanks for your contribution!

File: spec/requests/stories/articles_search_spec.rb [hunk 1]
```diff
@@ -6,5 +6,19 @@
       get "/search?q=hello"
       expect(response.body).to include("=> Search Results")
     end
+
+    context "with non-empty query" do
+      it "renders page with default page title" do
```
Comment (citizen428) [inline spec/requests/stories/articles_search_spec.rb:11]: Aren't the `it` descriptions reversed here? Isn't this one `"renders page with query in page title"` and the other one is `"renders page with default page title"`?

Comment (metamoni) [reply spec/requests/stories/articles_search_spec.rb:11]: You're right, my bad, thanks for catching that! I'm on it.

**Review by citizen428 — APPROVED** [review]

File: spec/requests/stories/articles_search_spec.rb (hunk 1 above)
Comment (metamoni) [reply spec/requests/stories/articles_search_spec.rb]: Having some `asdf` and ruby version issues locally that I can't seem to be able to fix 🤦🏻‍♀️ I made these changes via the web editor. Should be updated now.

**Review by citizen428 — APPROVED** [review]

**Comment by benhalpern** [comment]
@citizen428 @rhymes @nickytonline I want to flag that I believe this is going to cause a cache leak.

The search landing page sets cache control headers, so the whole page is cached and ignores the `q` parameter, so it returns plain old HTML.

Basically this is done so that the initial response can be fast and so the initial page load can withstand as much traffic as is thrown its way...

So this has to either be set on the client _or_ we remove this line from the controller:

`app/controllers/stories/articles_search_controller.rb` line 3:

```
    before_action :set_cache_control_headers, only: :index
```

And then add `q` to the list of safe params:

`config/fastly/snippets/safe_params_list.vcl`... (This only applies to Fastly-cached Forems).

Removing the cache headers seems fine to me. Shouldn't cause a major disruption.

Just like how we alert developers and break tests when they are using `current_user` in a place where it would cause a cache leak, we may want to do something similar with ignored params. We had a lot of problems like this before that automatic check went in. This functionality is in place so that random params won't generally cause cache misses... But in places like this it can cause problems.

**Comment by rhymes** [comment]
@benhalpern should we revert it in the meantime?

**Pull request merged**